app/src/firebase/users/GetSnappables.ts

```diff
@@ -0,0 +1,27 @@
+import firebase from "firebase/app";
+import type Snappable from "../../types/Snappable";
+
+/* Gets a list of Nicknames that are unique */
+const GetSnappables = async () => {
+    return new Promise(function (res, rej) {
+        const snappables = firebase
+            .firestore()
+            .collection("snappables")
+            .get({ source: "server" })
+            .then((querySnapshot) => {
+                let result: Snappable[] = [];
+                querySnapshot.forEach((doc) => {
+                    const { email, fullName, username } = doc.data();
+                    result.push({ id: doc.id, email, fullName, username });
+                });
+
+                return res(result);
+            })
+            .catch((e) => {
+                console.log("Major error in GetSnappables.");
+                rej(null);
+            });
```

Comment:
If I'm not mistaken the effect of this `catch` is to swallow the underlying exception `e` and only present a generic message to the user. This will make debugging challenging!

Reply:
This code is from other people's fork, I don't want to edit it in case it messes stuff up on their fork